### DISCUSSION TITLE
How to get data via web scraping if it has the same HTML format

### DISCUSSION THREAD

**Discussion by MF-Faqih** [teamalgoritma/community#670] [Q&A] [answered]
Selamat malam bapak/ibu semua, saya ingin bertanya cara mengambil informasi melalui web scraping jika dua data yang ingin diambil memiliki format HTML yang sama, beriku potongan kode yang saya gunakan:

    for page in range(1, 2):
        url_get = requests.get(url+str(page), headers = headers)
        soup = BeautifulSoup(url_get.text, 'html.parser')
        items = soup.findAll('div', 'row ListingCell-row ListingCell-agent-redesign')
        for item in items:
            try : name = item.find('h2', 'ListingCell-KeyInfo-title').text.strip()
            except : name = ''
            try : address = item.find('span', 'ListingCell-KeyInfo-address-text').text.strip()
            except : address = ''
            try : price = item.find('span', 'PriceSection-FirstPrice').text.strip().replace('Rp', '').replace('.', ',')
            except : price = ''
            bedroom = item.find('span', class_='KeyInformation-value_v2 KeyInformation-amenities-icon_v2').text.strip()
            print(bedroom )

sampai disini saya bisa mendapatkan informasi name, address, price dan bedroom, namun jika saya tambahkan line:

        bathroom = item.find('span', class_='KeyInformation-value_v2 KeyInformation-amenities-icon_v2').text.strip()

hasil dari print(bathroom) adalah list kosong. Bagaimana caranya untuk mengambil informasi dari dua item berbeda jika memiliki format HTML yang sama? Berikut saya sertakan beberapa gambar untuk mempermudah.

![33](https://github.com/teamalgoritma/community/assets/105436454/90c74160-0ff1-4285-88cd-b2d8c5b440a1)

*informasi yang ingin saya ambiladalah 1 (bedroom) dan 2 (bathroom)

Terimakasih banyak

**Comment by rusdipermana** [comment, marked as answer]
Mungkin bisa dibantu dengan mengambil kedua nilai dari elemen dengan kelas "KeyInformation-value_v2 KeyInformation-amenities-icon_v2", dapat menggunakan metode find_all() pada objek BeautifulSoup untuk menemukan semua elemen yang sesuai dengan kriteria tersebut. Kemudian,  dapat mengakses nilai-nilai tersebut menggunakan indeks.

Berikut adalah contoh kode untuk mengambil kedua nilai tersebut:

```
bedroom_elements = item.find_all('span', class_='KeyInformation-value_v2 KeyInformation-amenities-icon_v2')
if len(bedroom_elements) >= 2:
    bedroom_value1 = bedroom_elements[0].text.strip()
    bedroom_value2 = bedroom_elements[1].text.strip()
    print(bedroom_value1)
    print(bedroom_value2)
```

Dalam contoh di atas, bedroom_elements akan berisi semua elemen <span> yang memiliki kedua kelas yang diberikan. Kemudian, kita memeriksa apakah jumlah elemen tersebut cukup (minimal 2) sebelum mengakses nilai-nilainya menggunakan indeks. Anda dapat menyesuaikan kode di atas sesuai dengan kebutuhan. intinya kita memanfaatkan panjang dari indexnya dari element yang sama, semoga membantu :)

**Reply by MF-Faqih** [reply]
Terimakasih banyak, sudah berhasil tanpa dihasilkan error, semoga tuhan balas kebaikan Bapak :)
